### ISSUE TITLE
Enhancement: Generate correct GPX markup for track, route and POI

### ISSUE THREAD

**Issue by Friptick** [osmandapp/Osmand#9072]
[The GPX specification is quite clear about semantics.](https://en.wikipedia.org/wiki/GPS_Exchange_Format) Broadly:

- **Tracks** are made of **trackpoints**, they concern **the past** and should look like this: `<trk><trkseg><trkpt></trkpt>...</trkseg></trk>`
- **Routes** are made of **routepoints**, they concern **the future** and should look like this: `<rte><rtept></rtept>...</rte>`
- **POIs** or **favorites** are also known confusingly as **waypoints**, they are miscellaneous locations and can just be thrown in like this: `<wpt></wpt>`

Osmand currently _reads and displays_ these different types of data somewhat correctly when they are imported from elsewhere. But frustratingly it does not write them correctly itself:

- Tracks, or maybe they are supposed to be routes, can be exported from "markers" but they are marked up as generic POIs using `<wpt></wpt>`
- POIs are written to a file as vanilla "favorites" without problem

The solution is therefore to make a distinction between these 3 basic types of GPX file, so that "Add waypoint", or similar, will generate the right markup when used on a given file. The developers will know how best to implement this in practice. Thanks for noting this issue.

Closing #7473 and #8044 because they are going nowhere and this is the most obvious way forward.

**Comment by vshcherb** [comment]
OsmAnd uses all 3 types for last 10 years according to most usable concept:
1. Wpt - points displayed on the map
2. Track, trkpt - line visible on the map
3. RoutePoint - special points of the route, the route should be sometimes calculated between the se points. <rte> and <trkpt> could be used in combination where <rte> will display special points of the route such as turns.

**Comment by Friptick** [comment]
@vshcherb To repeat myself:

>Osmand currently _reads and displays_ these different types of data somewhat correctly when they are imported from elsewhere. But frustratingly it does not _write_ them correctly itself

Yes, Osmand **uses** (reads, displays) these data types, as you note.

No, it does not **write** them all. It does not write `<trkpt>`s or `<rtept>`s. It just writes `<wpt>`s for everything.

Forgive my possible ignorance, but I have tried everything. What am I not understanding?

**Comment by vshcherb** [comment]
If you try Measure Distance (Measurement tool) you will create a track within OsmAnd with <trkpt> or <rtept> or both at the same time. Later you can add <wpt> via context menu if you need to this created track.

**Comment by vshcherb** [comment]
I think there was misunderstanding but now you will be able to create any track you want within OsmAnd (except you might not be able to modify the name of <rtept>)

**Comment by Friptick** [comment]
@vshcherb Yes, I did try the measurement tool in the 8 years I have been trying to work around this problem with Osmand. I stand corrected that it generates `<trkpt>`. But it is unsatisfactory:

1. you cannot simply **add a place**, i.e. a town or existing favorite, to the end of a GPX track. You have to begin a new measurement line, from an arbitrary new place on the map. The resulting GPX markup is a bunch of `<trkseg>`s that don't join up.
2. there are **no timestamps** in the generated markup, unlike with the points added with `Add GPX waypoint` to tracks stored in the "tracks" directory. But those are not marked up as `<trkpt>` but rather as generic `<wpt>`, as if they were favorites with timestamps. This makes no sense and seems like a simple oversight, hence this issue.

A quick recap of what I want to do. It is a basic travel workflow: (1) Visit a town or favorite. (2) Add it to the GPX track. (3) Visit a new place, repeat, repeat, repeat for the whole trip. (4) Have a GPX file with a simple succession of places I have been with the dates I was there.

This seems like the most obvious use case for travelers. But you **cannot** do it in Osmand, see above. So I plan to post this as a new, more precise, issue. But since I would prefer you not to close it, can you please first confirm I have got nothing _factually_ wrong here?

**Comment by vshcherb** [comment]
Yes, now it becomes more clear and it is not possible todo today. We plan to extend our Markers functionality linked to GPX, so later we will record when Marker was marked as passed into GPX as visited with timestamp.

Though OsmAnd doesn't all possible features of GPX to edit and actually there is no master plan for this, so probably you need to find another app for that.

**Comment by Friptick** [comment]
@vshcherb Okay, thanks for that. Please just note that this is a sort-of-bug which has a quick fix, not involving Markers:

**Just use `<trk>` markup for what is called a "Track" in the interface and stored in a "tracks" directory in Android. Because that is what `<trk>` is for.**

This would solve the issue. The created Tracks already have timestamps, They just have the wrong XML.
